server/src/main/java/org/opensearch/plugins/SecureTransportSettingsProvider.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.plugins;
+
+import org.opensearch.common.annotation.ExperimentalApi;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.http.HttpServerTransport;
+import org.opensearch.transport.TcpTransport;
+
+import javax.net.ssl.SSLEngine;
+import javax.net.ssl.SSLException;
+
+import java.util.Optional;
+
+/**
+ * A provider for security related settings for transports.
+ *
+ * @opensearch.experimental
+ */
+@ExperimentalApi
+public interface SecureTransportSettingsProvider {
+    /**
+     * An exception handler for errors that might happen while secure transport handle the requests.
+     *
+     * @opensearch.experimental
+     */
+    @ExperimentalApi
+    interface ServerExceptionHandler {
+        static ServerExceptionHandler NOOP = new ServerExceptionHandler() {
```

Comment:
Style nitpick, but can you implement this as `t -> {}` without all the boilerplate?